zaino-serve/src/rpc/service.rs

```diff
@@ -816,17 +831,39 @@ impl CompactTxStreamer for GrpcClient {
     {
         println!("[TEST] Received call of get_taddress_balance.");
         Box::pin(async {
-            Err(tonic::Status::unimplemented("get_taddress_balance not yet implemented. If you require this RPC please open an issue or PR at the Zingo-Indexer github (https://github.com/zingolabs/zingo-indexer)."))
+            let zebrad_client = JsonRpcConnector::new(
+                self.zebrad_uri.clone(),
+                Some("xxxxxx".to_string()),
+                Some("xxxxxx".to_string()),
+            )
+            .await?;
+            let taddrs = request.into_inner().addresses;
+            if !taddrs.iter().all(|taddr| check_taddress(taddr).is_some()) {
```

Comment:
i havent doubled check this is the same logic but is this simpler?
```suggestion
            if taddrs.iter().any(|taddr| check_taddress(taddr).is_none()) {
```